pkg/operator/operator.go

```diff
@@ -606,3 +588,63 @@ func (o *Operator) setMutatingWebhookCABundle(ctx context.Context, caBundle []by
 	}
 	return o.client.Update(ctx, &mwc)
 }
+
+func (o *Operator) canUpdateWebhooks(ctx context.Context) (bool, error) {
+	return o.canI(ctx,
+		authv1.ResourceAttributes{
+			Group:     arv1.GroupName,
+			Resource:  "MutatingWebhookConfiguration",
+			Namespace: o.opts.OperatorNamespace,
+			Verb:      "update",
+		},
+		authv1.ResourceAttributes{
+			Group:     arv1.GroupName,
+			Resource:  "ValidatingWebhookConfiguration",
+			Namespace: o.opts.OperatorNamespace,
+			Verb:      "update",
+		},
+	)
+}
+
+func (o *Operator) canI(ctx context.Context, resources ...authv1.ResourceAttributes) (bool, error) {
+	for _, resource := range resources {
+		req := authv1.SelfSubjectAccessReview{
+			Spec: authv1.SelfSubjectAccessReviewSpec{
+				ResourceAttributes: &resource,
+			},
+		}
+		if err := o.client.Create(ctx, &req); err != nil {
+			return false, fmt.Errorf("check permissions to %q %q: %w", resource.Verb, resource.Name, err)
+		}
+		if !req.Status.Allowed {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func (o *Operator) continuouslySetCABundle(ctx context.Context, caBundle []byte) {
```

Comment:
**note:** link to non-polling solution for this: https://github.com/GoogleCloudPlatform/prometheus-engine/pull/350